packages/react/src/components/select/custom/custom-select.tsx

```diff
@@ -31,39 +32,46 @@ export function CustomSelect({
   onClick,
   onKeyUp,
   onOpenChange,
+  value,
   ...restProps
 }: CustomSelectProps) {
   const selectRef = useRef<HTMLSelectElement>(null);
-  const [selectedOption, setSelectedOption] = useState<ReactNode>();
-  const [value, setValue] = useState(restProps.value ?? defaultValue);
+  const options = useRef(new Map<typeof value, ReactNode>());
+  const [currentValue, setCurrentValue] = useState<typeof value>();
+
+  // default to first option after first render
```

Comment:
I think this is more accurate:

```suggestion
  // default to first option on first render
```